transmission/transmission.go

```diff
@@ -538,21 +557,48 @@ func (b *batchAgg) enqueueErrResponses(err error, events []*Event, duration time
 	}
 }
 
+var zstdBufferPool sync.Pool
+
+type pooledReader struct {
+	*bytes.Reader
+	buf []byte
+}
+
+func (r *pooledReader) Close() error {
+	zstdBufferPool.Put(r.buf[:0])
+	r.Reader = nil
+	r.buf = nil
+	return nil
+}
+
+// Instantiating a new encoder is expensive, so use a global one.
+// The docs say EncodeAll() is concurrency-safe.
+var zstdEncoder *zstd.Encoder
+
+func init() {
+	var err error
+	zstdEncoder, err = zstd.NewWriter(nil, zstd.WithEncoderLevel(zstd.EncoderLevelFromZstd(2)))
```

Comment:
have you benchmarked which encoder level is right for us?

Reply:
Not with this implementation! I will do so.

Reply:
I re-tested this against gzip "fastest" compression. With this setting zstd gets slightly better compression in a little more than half the time. The compression is a little worse than gzip default, which is ok.

Reply:
For our purposes, I think we'd like better compression as long as performance is reasonable enough. Network cost is a bigger component of our cost & our customers' cost than CPU I think? especially if they're paying 2 to 8 cents per GB to talk to us?

Reply:
I imagine it varies with the input data, but for the test I've got here we have to double (or more) the compute time to get an extra 1-2% compression.

Reply:
Okeydoke!